contracts/modules/metadata/CoreMetadataModule.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { IIPAccount } from "../../interfaces/IIPAccount.sol";
+import { AccessControlled } from "../../access/AccessControlled.sol";
+import { BaseModule } from "../BaseModule.sol";
+import { Errors } from "../../lib/Errors.sol";
+import { IPAccountStorageOps } from "../../lib/IPAccountStorageOps.sol";
+import { CORE_METADATA_MODULE_KEY } from "../../lib/modules/Module.sol";
+import { ICoreMetadataModule } from "../../interfaces/modules/metadata/ICoreMetadataModule.sol";
+
+/// @title CoreMetadataModule
+/// @notice Manages the core metadata for IP assets within the Story Protocol, all metadata can only update once.
+/// @dev This contract allows setting core metadata attributes for IP assets.
+///      It implements the ICoreMetadataModule interface.
+contract CoreMetadataModule is BaseModule, AccessControlled, ICoreMetadataModule {
+    using IPAccountStorageOps for IIPAccount;
+
+    string public override name = CORE_METADATA_MODULE_KEY;
+
+    /// @notice Modifier to ensure that metadata can only be set once.
+    modifier onlyOnce(address ipAccount, bytes32 metadataName) {
+        if (!_isEmptyString(IIPAccount(payable(ipAccount)).getString(metadataName))) {
+            revert Errors.CoreMetadataModule__MetadataAlreadySet();
+        }
+        _;
+    }
+
+    /// @notice Creates a new CoreMetadataModule instance.
+    /// @param accessController The address of the AccessController contract.
+    /// @param ipAccountRegistry The address of the IPAccountRegistry contract.
+    constructor(
+        address accessController,
+        address ipAccountRegistry
+    ) AccessControlled(accessController, ipAccountRegistry) {}
+
+    /// @notice Sets the name for an IP asset.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
+    /// @param ipAccount The address of the IP asset.
+    /// @param ipName The name to set for the IP asset.
+    function setIpName(address ipAccount, string memory ipName) external verifyPermission(ipAccount) {
+        _setIpName(ipAccount, ipName);
+    }
+
+    /// @notice Sets the description for an IP asset.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
+    /// @param ipAccount The address of the IP asset.
+    /// @param description The description to set for the IP asset.
+    function setIpDescription(address ipAccount, string memory description) external verifyPermission(ipAccount) {
```

Comment:
What's the use of description onchain? Let's follow ERC721 and 1155's metadata approach, have name and tokenURI but not description. I can't think of any onchain applications using the description of another contract.

Reply:
in beta, the description returned within TokenURI of IPAsset. but with the new metadata spec for mainnet discussion, we decided to remove the "description" metadata.